libs/commons/Testutil_files.mli

```diff
@@ -0,0 +1,63 @@
+(*
+   Utilities for creating, scanning, and deleting a hierarchy
+   of test files.
```

Comment:
maybe an example of use would be nice, like an example of test using this API.

Note that at first because of the name (Testutil_files.mli), I thought it was creating fake files, in a virtual
filesystem, so maybe it's worth saying this module will create real files.

Also because it creates real files, what's the advantage of your API vs using directly the Unix or Sys
API to create files?
Maybe worth a comment too.